openmls/src/messages/proposals.rs

```diff
@@ -200,29 +182,22 @@ impl From<ProposalType> for u16 {
 /// } Proposal;
 /// ```
 #[allow(clippy::large_enum_variant)]
-#[derive(Debug, PartialEq, Clone, Serialize, Deserialize, TlsSize, TlsSerialize)]
+#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
 #[allow(missing_docs)]
 #[repr(u16)]
 pub enum Proposal {
-    #[tls_codec(discriminant = 1)]
     Add(AddProposal),
-    #[tls_codec(discriminant = 2)]
     Update(UpdateProposal),
-    #[tls_codec(discriminant = 3)]
     Remove(RemoveProposal),
-    #[tls_codec(discriminant = 4)]
     PreSharedKey(PreSharedKeyProposal),
-    #[tls_codec(discriminant = 5)]
     ReInit(ReInitProposal),
-    #[tls_codec(discriminant = 6)]
     ExternalInit(ExternalInitProposal),
-    #[tls_codec(discriminant = 7)]
     GroupContextExtensions(GroupContextExtensionProposal),
     // # Extensions
     // TODO(#916): `AppAck` is not in draft-ietf-mls-protocol-17 but
     //             was moved to `draft-ietf-mls-extensions-00`.
-    #[tls_codec(discriminant = 8)]
     AppAck(AppAckProposal),
+    Other((u16, OtherProposal)),
```

Comment:
Why wouldn't we align this with `CustomProposal`? And the other question is whether we want a wrapper struct instead of a tuple. I don't have a strong opinion but we have not been using tuples a lot and they are less expressive in general.